### PR TITLE
fix: correct behaviors around single-target watch apps vs watch extensions to avoid crashes when targeting watchOS 8 and lower

### PR DESCRIPTION
**Pull request by thedavidharris**
### Checklist

* [ ] Does your PR title have the correct [title format](https://github.com/amplitude/Amplitude-SDK-Template/blob/main/CONTRIBUTING.md#pr-commit-title-conventions)?
* Does your PR have a breaking change?:  No

### CODE REVIEW

File: Sources/Amplitude/Plugins/watchOS/WatchOSLifecycleMonitor.swift [hunk 1]
```diff
@@ -13,27 +13,57 @@
     class WatchOSLifecycleMonitor: UtilityPlugin {
         var wasBackgrounded: Bool = false
 
-        private var watchExtension = WKApplication.shared()
```
Comment (thedavidharris) [inline Sources/Amplitude/Plugins/watchOS/WatchOSLifecycleMonitor.swift:16]: This wasn't referenced anywhere

File: Sources/Amplitude/Plugins/watchOS/WatchOSLifecycleMonitor.swift [hunk 2]
```diff
@@ -13,27 +13,57 @@
...
+                case WKApplication.willEnterForegroundNotification:
+                    self.applicationWillEnterForeground(notification: notification)
+                case WKApplication.didEnterBackgroundNotification:
+                    self.applicationDidEnterBackground(notification: notification)
+                default:
+                    break
+                }
+            } else if !isSingleTargetApplication {
+                switch notification.name {
+                case WKExtension.applicationWillEnterForegroundNotification:
```
Comment (thedavidharris) [inline Sources/Amplitude/Plugins/watchOS/WatchOSLifecycleMonitor.swift:58]: Splitting this out is probably overkill but wanted to be extra safe around when WKApplication or WKExtension was referenced.

**Comment by thedavidharris** [comment]
Unsure why this test is failing.

**Comment by crleona** [comment]
@thedavidharris Thank you for the PR! I had started work on another request (https://github.com/amplitude/Amplitude-Swift/pull/180) which solves an additional issue we had with tracking sessions and uses slightly simpler logic (as both WKApplication and WKExtension notifications have the same underlying rawValues and acted the same across the watchOS versions I tested). Closing this PR, but please let me know if you have a scenario that I missed that require the isSingleTargetApplication / watchOS 9.0 checks.

**Pull request closed without merging**